wps/core/models.py

```diff
@@ -16,20 +16,92 @@
 """Defines dataclasses for business-logic data as well as request/reply models for use
 in the API."""
 
-from pydantic import BaseModel
+from dataclasses import dataclass
+from enum import Enum
 
+from ghga_service_commons.utils.utc_dates import DateTimeUTC
+from pydantic import BaseModel, EmailStr, Field
 
-class WorkPackageData(BaseModel):
+__all__ = [
+    "WorkType",
+    "WorkOrderToken",
+    "WorkPackageCreationData",
+    "WorkPackageData",
+    "WorkPackage",
+]
+
+
+class BaseDto(BaseModel):
+    """Base model preconfigured for use as Dto."""
+
+    class Config:  # pylint: disable=missing-class-docstring
+        extra = "forbid"
+        frozen = True
+
+
+class WorkType(str, Enum):
+    """The type of work that a work package describes."""
+
+    DOWNLOAD = "download"
+    UPLOAD = "upload"
+
+
+@dataclass(frozen=True)
+class WorkOrderToken:
+    """A class describing a work order token."""
+
+    type: WorkType
+    file_id: str
+    user_id: str
+    public_key: str
+    full_user_name: str
+    email: str
+
+
+class WorkPackageCreationData(BaseDto):
+    """
+    All data necessary to create a work package.
+    """
+
+    user_id: str = Field(default=..., title="ID od the user")
+    dataset_id: str = Field(default=..., title="ID of the dataset")
+    type: WorkType = Field(default=..., title="Work type")
+    file_ids: list[str] = Field(default=..., title="IDs of all included files")
+    public_key: str = Field(
+        default=...,
+        title="Crpyt4GH key of the user",
+        description="The user's public Crpyt4GH key in base64 encoding",
+    )
+
+
+class WorkPackageData(WorkPackageCreationData):
     """
-    All data necessary to describe a work package.
+    All data that describes a work package.
     """
 
-    name: str
+    full_user_name: str = Field(
```

Comment:
Here now we have to add the full list of file IDs from the dataset:
```suggestion
    file_ids: list[str] = Field(default=..., description="IDs of all included files")
    full_user_name: str = Field(
```

Reply:
Ok, I see, we can then make it optional in WorkPackageCreationData. Yes, maybe that's a bit more explicitly showing how this is used.